### ISSUE TITLE
`expect(nil).toAlways(equal(0))` incorrectly succeeds

### ISSUE THREAD

**Issue by defagos** [Quick/Nimble#1120]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/main/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

Consider the following expectation:

```swift
func testExpectNilToAlwaysEqualZero() {
    expect(nil).toAlways(equal(0))
}
````

You would expect a test failure but the test actually succeeds.

The issue is likely associated with polling since the following expectation fails as expected:

```swift
func testExpectNilToEqualZero() {
    expect(nil).to(equal(0))
}
````

### What did you expect to happen?

The expectation in `testExpectNilToAlwaysEqualZero()` above must fail.

### What actually happened instead?

The expectation in `testExpectNilToAlwaysEqualZero()` above incorrectly succeeds.

### Environment

- Nimble: 13.2.0
- Xcode Version: 15.2.0
- Swift Version: 5.9.2
- Swift Package Manager 5.9.0

**Comment by younata** [comment]
Thanks for identifying this!

**Comment by younata** [comment]
After looking into this more, this type of issue also applies to `toNever`. Lovely.

**Comment by defagos** [comment]
Thanks for your answer.

As a temporary workaround it is possible to force-unwrap potentially nullable values in related expectations so that mismatches at least do not get unnoticed:

```swift
expect(somePotentiallyNilValue!).toAlways(equal(1234))
```

**Comment by younata** [comment]
This is now fixed in [Nimble 13.2.1](https://github.com/Quick/Nimble/releases/tag/v13.2.1).

**Comment by defagos** [comment]
Thank you very much for the quick fix, really appreciated!
